wol/main.py

```diff
@@ -0,0 +1,184 @@
+
+    def __eq__(self, other):
+        return self.mac == other.mac
+
+    def to_bytes(self):
+        return copy.copy(self.mac)
+
+
+BROADCAST_MAC = MacAddress(b'\xff\xff\xff\xff\xff\xff')
+
+
+def is_root():
+    return os.geteuid() == 0
+
+
+def get_interface_operstate(interface):
+    with open('/sys/class/net/{}/operstate'.format(interface), 'r') as f:
+        return f.read().strip().lower()
+
+
+def get_interface_mac(interface):
+    mac = netifaces.ifaddresses(interface)[netifaces.AF_LINK][0].get('addr')
+    return MacAddress(binascii.unhexlify(mac.replace(':', '').replace('-', '')))
+
+
+def build_magic_packet(interface, target_mac, broadcast, password):
+    dst_mac = BROADCAST_MAC if broadcast else target_mac
+    src_mac = get_interface_mac(interface)
+    return dst_mac.to_bytes() + src_mac.to_bytes() + ETHER_TYPE_WOL \
+        + b'\xff' * 6 + target_mac.to_bytes() * 16 + password
+
+
+def send_magic_packet(interface, pkt, count, interval, verbose):
+    target_mac = MacAddress(pkt[20:26])
+    if verbose:
+        print("Sending {} magic packet to {} via interface {}".format(count, target_mac, interface))
+    sock = socket.socket(socket.AF_PACKET, socket.SOCK_RAW)
+    sock.bind((interface, 0))
+    for i in range(count):
+        sock.send(pkt)
+        if verbose:
+            print("{} magic packet sent to {}".format(ORDINAL_NUMBER[i + 1], target_mac))
+        if i + 1 != count:
+            time.sleep(interval / 1000)
+    sock.close()
+
+
+def validate_interface(ctx, param, value):
+    if value not in netifaces.interfaces():
+        raise click.BadParameter("invalid SONiC interface name {}".format(value))
+    if get_interface_operstate(value) != 'up':
+        raise click.BadParameter("interface {} is not up".format(value))
+    return value
+
+
+def validate_target_mac(ctx, param, value):
+    mac_list = []
+    for mac_s in value.split(','):
+        try:
+            mac_b = binascii.unhexlify(mac_s.replace(':', '').replace('-', ''))
+        except binascii.Error:
+            raise click.BadParameter("invalid MAC address {}".format(mac_s))
+        if len(mac_b) != 6:
+            raise click.BadParameter("invalid MAC address {}".format(mac_s))
+        mac_list.append(MacAddress(mac_b))
+    return mac_list
+
+
+def validate_password(ctx, param, value):
+    if len(value) == 0:
+        return b''  # Empty password is valid.
+    elif len(value) <= 15:  # The length of a valid IPv4 address is less or equal to 15.
+        try:
+            password = socket.inet_aton(value)
+        except OSError:
+            raise click.BadParameter("invalid password format")
+    else:  # The length of a valid MAC address is 17.
+        try:
+            password = binascii.unhexlify(value.replace(':', '').replace('-', ''))
+        except binascii.Error:
+            raise click.BadParameter("invalid password format")
+    if len(password) not in [4, 6]:
+        raise click.BadParameter("password must be 4 or 6 bytes")
```

Comment:
this statement is conflicting with line 128

Reply:
line 128 `return` directly, will not hit this line.

Reply:
I mean the wording. You said that *must be* 4 or 6, but 0 is also acceptable based on your implementation.

Reply:
I see. Updated.